source/opt/ir_context.cpp

```diff
@@ -412,7 +405,8 @@ void IRContext::KillOperandFromDebugInstructions(Instruction* inst) {
         continue;
       auto& operand = it->GetOperand(kDebugGlobalVariableOperandVariableIndex);
       if (operand.words[0] == id) {
-        operand.words[0] = GetOpenCL100DebugInfoNone()->result_id();
+        operand.words[0] =
+            get_debug_info_mgr()->GetDebugInfoNone()->result_id();
```

Comment:
Are we sure that the DebugInfoNone will be defined before |inst|, or are forward references allowed here?

Reply:
Oh nice catch. If the input SPIR-V has `DebugInfoNone` after `DebugFunction` or `DebugGlobalVariable`, it causes illegal forward references. I updated `DebugInfoManager` so that it moves the exisitng `DebugInfoNone` to the beginning of the debug instruction list.